apis/rng/src/lib.rs

```diff
@@ -0,0 +1,85 @@
+#![no_std]
+
+use core::cell::Cell;
+
+use libtock_platform::{share, AllowRw, DefaultConfig, ErrorCode, Subscribe, Syscalls};
+pub struct Rng<S: Syscalls>(S);
+
+impl<S: Syscalls> Rng<S> {
+    /// Returns Ok() if the driver was present.This does not necessarily mean
+    /// that the driver is working.
+    pub fn exists() -> Result<(), ErrorCode> {
+        S::command(DRIVER_NUM, EXISTS, 0, 0).to_result()
+    }
+
+    /// Register a listener to be called when the random generation is finished
+    pub fn register_listener<'share>(
+        listener: &'share Cell<Option<(u32, u32)>>,
+        subscribe: share::Handle<Subscribe<'share, S, DRIVER_NUM, 0>>,
+    ) -> Result<(), ErrorCode> {
+        S::subscribe::<_, _, DefaultConfig, DRIVER_NUM, 0>(subscribe, listener)
+    }
+
+    /// Sets the buffer in which the random numbers will be written
+    pub fn set_buffer<'share>(
+        buffer: &'share mut [u8],
+        allow_rw: share::Handle<AllowRw<'share, S, DRIVER_NUM, RW_ALLOW>>,
+    ) -> Result<(), ErrorCode> {
+        S::allow_rw::<DefaultConfig, DRIVER_NUM, RW_ALLOW>(allow_rw, buffer)
+    }
+
+    /// Initiates an async random generation for n bytes
+    /// A buffer and a callback should have been set before
+    pub fn get_random(n: u32) -> Result<(), ErrorCode> {
+        S::command(DRIVER_NUM, ASK_FOR_RANDOM_BYTES, n, 0).to_result()
+    }
+
+    /// Initiates a synchronous random number generation
+    /// `n` random bytes will be written in `buf`
+    /// if n > buf.len() then only buf.len() bytes will be written in buf
```

Comment:
I would return a `SIZE` error instead.

Reply:
So I should store the size of the previously set buffer and check this also for the async function, right? It doesn't seem right for the sync and async functions to have a different approach on this.

Reply:
I made the change on the sync function, but I don't know how or if it is possible to do it also for the async one.